code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -464,3 +434,335 @@
+
+	game_plane_master_controller.add_filter("saturnx_filter", 10, color_matrix_filter(col_filter_twothird, FILTER_COLOR_HCY))
+
+	for(var/filter in game_plane_master_controller.get_filters("saturnx_filter"))
+		animate(filter, loop = -1, color = col_filter_full, time = 4 SECONDS, easing = CIRCULAR_EASING|EASE_IN, flags = ANIMATION_PARALLEL)
+		//uneven so we spend slightly less time with bright colors
+		animate(color = col_filter_twothird, time = 6 SECONDS, easing = LINEAR_EASING)
+		animate(color = col_filter_half, time = 3 SECONDS, easing = LINEAR_EASING)
+		animate(color = col_filter_empty, time = 2 SECONDS, easing = CIRCULAR_EASING|EASE_OUT)
+		animate(color = col_filter_half, time = 24 SECONDS, easing = CIRCULAR_EASING|EASE_IN)
+		animate(color = col_filter_twothird, time = 12 SECONDS, easing = LINEAR_EASING)
+
+	game_plane_master_controller.add_filter("saturnx_blur", 1, list("type" = "radial_blur", "size" = 0))
+
+	for(var/filter in game_plane_master_controller.get_filters("saturnx_blur"))
+		animate(filter, loop = -1, size = 0.04, time = 2 SECONDS, easing = ELASTIC_EASING|EASE_OUT, flags = ANIMATION_PARALLEL)
+		animate(size = 0, time = 6 SECONDS, easing = CIRCULAR_EASING|EASE_IN)
+
+///This proc turns the guy who took the drug invisible by giving him the invisible man trait and updating his body, this changes the sprite of all his organic limbs to a 1 alpha version.
+/datum/reagent/drug/saturnx/proc/turn_man_invisible(mob/living/carbon/invisible_man)
+	if(!invisible_man.getorganslot(ORGAN_SLOT_LIVER))
+		return
+	if(invisible_man.undergoing_liver_failure())
+		return
+	if(HAS_TRAIT(invisible_man, TRAIT_NOMETABOLISM))
+		return
+
+	ADD_TRAIT(invisible_man, TRAIT_INVISIBLE_MAN, name)
+
+	var/datum/dna/druggy_dna = invisible_man.has_dna()
+	if(druggy_dna?.species)
+		druggy_dna.species.species_traits += NOBLOODOVERLAY
+
+	invisible_man.update_body()
+	invisible_man.update_hair()
+	invisible_man.remove_from_all_data_huds()
+	invisible_man.sound_environment_override = SOUND_ENVIROMENT_PHASED
+
+/datum/reagent/drug/saturnx/on_mob_end_metabolize(mob/living/M)
+	. = ..()
+	if(HAS_TRAIT(M, TRAIT_INVISIBLE_MAN))
+		M.add_to_all_human_data_huds() //Is this safe, what do you think, Floyd?
+		REMOVE_TRAIT(M, TRAIT_INVISIBLE_MAN, name)
+		to_chat(M, span_notice("As you sober up, opacity once again returns to your body meats."))
+
+		var/datum/dna/druggy_dna = M.has_dna()
+		if(druggy_dna?.species)
+			druggy_dna.species.species_traits -= NOBLOODOVERLAY
+
+	M.update_body()
+	M.update_hair()
+	M.sound_environment_override = NONE
+
+	if(!M.hud_used)
+		return
+
+	var/atom/movable/plane_master_controller/game_plane_master_controller = M.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
+	game_plane_master_controller.remove_filter("saturnx_filter")
+	game_plane_master_controller.remove_filter("saturnx_blur")
+
+/datum/reagent/drug/saturnx/overdose_process(mob/living/M, delta_time, times_fired)
```

Comment:
please name `M` better

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names